API/Array.hpp.in

```diff
@@ -0,0 +1,29 @@
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

Comment:
True. Having these typedefs/usings set up in separate files will facilitate things when we want to replace them later on. Everything is set up as if they were concrete classes.
The unit library will be similar in setup but on a larger scale.